tatsu/walkers.py

```diff
@@ -8,10 +8,18 @@
 from tatsu.util import is_list
 
 
-class NodeWalker:
-    def __new__(cls, *args, **kwargs):
-        cls._walker_cache = {}
-        return super(NodeWalker, cls).__new__(cls)
+class NodeWalkerMeta(type):
+    def __new__(cls, name, bases, dct):
+        class_ = super().__new__(cls, name, bases, dct)
+        class_._walker_cache = {}
+        return class_
+
+
+class NodeWalker(metaclass=NodeWalkerMeta):
+    def __init__(self):
+        super().__init__()
+        # copy the class attribute to avoid linter warnings
+        self._walker_cache = type(self)._walker_cache  # pylint: disable=no-member
```

Comment:
How is this any different than the simpler `self._walker_cache = {}` ?

Reply:
Please disregard the previous comment, I was confused by the comment in the code. The cache dictionary is not copied.

Reply:
Please elaborate? Which do you think should be the text of the comment?

Reply:
Reading the comment in the code, I though that what the code did was something like this:
```python
self._walker_cache = type(self)._walker_cache.copy()
```
I don't think the comment is wrong, it is just that the code pattern is a very unusual and I would prefer if there would be another way to silence the pylint warnings.

Reply:
I think that the patch doesn't hurt, it is inexpensive, and it silences all the `pylint` warnings without changing the existing code.